app/src/org/commcare/dalvik/odk/provider/FormsProvider.java

```diff
@@ -106,9 +119,12 @@ public boolean onCreate() {
     }
     
     public void init() {
-        //this is terrible, we need to be binding to the cc service, etc. Temporary code for testing
-        if(mDbHelper == null) {
-            mDbHelper = new DatabaseHelper(CommCareApplication._(), DATABASE_NAME);
+        // this is terrible, we need to be binding to the cc service, etc. Temporary code for testing
```

Comment:
...This doesn't seem like it ended up being temporary code. Has this turned out to be acceptable, or should we be changing it?